test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java

```diff
@@ -1292,6 +1299,56 @@ public Environment newEnvironment(Settings settings) {
         return TestEnvironment.newEnvironment(build);
     }
 
+    /**
+     * Create a new {@link ContextIndexSearcher} from the given reader, randomizing if an executor for
+     * multi-threaded collection will be used or not
+     */
+    public ContextIndexSearcher newContextSearcher(IndexReader reader) throws IOException {
```

Comment:
There is a newINdexSearcher method in AggregatorTestCase that seems like it's doing a very similar thing, maybe the two can be merged into one?